src/ol_infrastructure/providers/salt/minion.py

```diff
@@ -0,0 +1,104 @@
+"""
+Dynamic provider that uses the SaltStack API to provision a minion.
+
+Registers a minion ID with a SaltStack master and generates a keypair.  The keypair is returned so that it can be passed
+to an instance via user data to take advantage of cloud-init.
+"""
+from typing import Optional, Text
+
+from pepper import Pepper
+from pulumi import Input, Output, ResourceOptions
+from pulumi.dynamic import CreateResult, ReadResult, Resource, ResourceProvider
+from pydantic import BaseModel, SecretStr
+
+
+class OLSaltStackInputs(BaseModel):
+    minion_id: Input[Text]
+    salt_api_url: Input[Text]
+    salt_user: Input[Text]
+    salt_password: Input[SecretStr]
+    salt_auth_method: Input[Text] = 'pam'
+
+
+class _OLSaltStackProviderInputs(BaseModel):
```

Comment:
Does this need to be its own class or can those attributes be part of the `OLSaltStackProvider` one?

Reply:
It needs to be its own class, because these are the 'unwrapped' values, where the other class will be receiving the data as a Pulumi `Input` type. https://www.pulumi.com/docs/intro/concepts/programming-model/#resource-provider-interface